app/controllers/application_controller.rb

```diff
@@ -13,5 +14,13 @@ def current_user
     User.find(session[:user_id]) rescue nil
   end
 
-  helper_method :cart, :current_user
+  def previous_path
+    session[:previous_path]
+  end
+
+  def set_previous_path
+    session[:previous_path] = request.env["HTTP_REFERER"].gsub(/\w*\:\/\/[^\/]+/, '') rescue nil
```

Comment:
I think that we can use like "path" or something from require URI. Instead of the regex here.